functorch/_src/aot_autograd.py

```diff
@@ -409,14 +417,21 @@ def forward(ctx, *deduped_flat_tensor_args):
                 CompiledFunction.compiled_fw, deduped_flat_tensor_args
             )
             num_outs = CompiledFunction.num_outs
-            ctx.save_for_backward(*fw_outs[num_outs:])
+            num_symints = CompiledFunction.num_symints
+            if num_symints > 0:
+                ctx.save_for_backward(*fw_outs[num_outs:-num_symints])
```

Comment:
What guarantees that the symints are the suffix of the graph in question?

Reply:
this is one of the gaps i need to figure this out better before i land it. i couldn't figure out a test case that violated this, but i also didn't prove it's true.

Reply:
The graph you are partitioning here is generated by joint graph in FX. So that's the code to audit. But I don't see why it would organize the outputs this way. Maybe the sizes are getting shunted to the end because of lazy tracing, but if so, that's a very delicate way the invariant is coming up and we will need to handle it explicitly. Maybe @Chillee knows.

Reply:
@wconstab No I don't think it's true that they're always at the end. They should be ordered by their relative position in the original graph: https://github.com/pytorch/pytorch/blob/master/functorch/_src/partitioners.py#L392-L393

Although activations are always after the original outputs.

Reply:
i think i fixed this now. pls take a look

Reply:
The logic... looks the same? What am I looking at